src/main/java/org/quiltmc/loader/impl/discovery/ModResolver.java

```diff
@@ -326,6 +475,35 @@ private static IVecInt toVecInt(IntStream stream) {
 		return result;
 	}
 
+	void processDependencies(Logger logger, Map<String, ModIdDefinition> modDefs, DependencyHelper<LoadOption,
+		ModLink> helper, ModCandidate mc, ModLoadOption option)
+		throws ContradictionException {
+
+		for (ModDependency dep : mc.getInfo().getDepends()) {
+			ModIdDefinition def = modDefs.get(dep.getModId());
+			if (def == null) {
+				def = new OptionalModIdDefintion(dep.getModId(), new ModLoadOption[0]);
+				modDefs.put(dep.getModId(), def);
+				def.put(helper);
+			}
+
+			new ModDep(logger, option, dep, def).put(helper);
+		}
+
+		for (ModDependency conflict : mc.getInfo().getBreaks()) {
+			ModIdDefinition def = modDefs.get(conflict.getModId());
+			if (def == null) {
+				def = new OptionalModIdDefintion(conflict.getModId(), new ModLoadOption[0]);
+				modDefs.put(conflict.getModId(), def);
+
+				def.put(helper);
+			}
+
+			new ModBreakage(logger, option, conflict, def).put(helper);
+		}
+	}
+
+	// TODO: Convert all these methods to new error syntax
```

Comment:
No.

Currently quilt-loader has two error generation paths: the "simple" one (where there are no jij mods, and as such all mods found must be loaded), and the "advanced" one (where we have jij mods and must use sat4j to find a valid mod set). All of those methods are for generating the error messages for the "simple" type. Ideally, I'd like to unify all error message generation, although as I haven't written proper message generation for "advanced" yet (and don't plan to for this PR) I assume this would be left uncompleted.